src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -3106,6 +3106,24 @@ OGRErr QgsOgrLayer::GetExtent( OGREnvelope *psExtent, bool bForce )
   return OGR_L_GetExtent( hLayer, psExtent, bForce );
 }
 
+OGRErr QgsOgrLayer::GetExtent3D( OGREnvelope3D *psExtent3D, bool bForce )
+{
+  QMutexLocker locker( &ds->mutex );
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,9,0)
+  return OGR_L_GetExtent3D( hLayer, psExtent3D, bForce );
+#else
+  OGREnvelope psExtent;
```

Comment:
you should probably fallback to compute3DExtentSlowly() to get same results in QGIS whatever the GDAL version

Reply:
done